ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -1681,6 +1681,9 @@ boolean doPhase1(ASTNode ast, QB qb, Phase1Ctx ctx_1, PlannerContext plannerCtx)
               && ch.getChild(0) instanceof ASTNode) {
             ch = (ASTNode) ch.getChild(0);
             isTmpFileDest = ch.getToken().getType() == HiveParser.TOK_TMP_FILE;
+            if (ch.getToken().getType() == HiveParser.StringLiteral) {
```

Comment:
Could this be something else?

Reply:
I don't think so. As per my understanding "DIRECTORY" is always followed by string literal and this happens only for INSERT query.

Reply:
Then, I think this should be an assert.

Reply:
why assert? The code block inside the condition `if (ast.getChildCount() > 0 && ast.getChild(0) instanceof ASTNode) {
          ASTNode ch = (ASTNode) ast.getChild(0);`  is also accessed by other queries for example queries which have `HiveParser.TOK_TMP_FILE  `

Reply:
@pvary 
For Example : `ANALYZE TABLE src COMPUTE STATISTICS FOR COLUMNS key,value` such queries will hit the above condition and having assert will break it right? Is my understanding correct or am i missing anything?

Reply:
What will be the value for `ch.getToken().getType()` in this case?

Reply:
Ahh, so it could be something else then 😄 
That explains this line, thanks